src/main/java/org/jenkinsci/plugins/workflow/cps/CpsThread.java

```diff
@@ -183,7 +176,7 @@ public StepExecution getStep() {
             LOGGER.log(FINE, "runNextChunk on {0}", resumeValue);
             final Outcome o = resumeValue;
             resumeValue = null;
-            outcome = program.run0(o, categories);
+            outcome = program.run0(o, getCategories());
```

Comment:
Instead of having to recreate the list over and over, maybe you could replace the static initialization block with something like this instead?
```
private static final List<Class> categories = ImmutableList.builder().addAll(Continuable.categories).add(IteratorHack.class).build();
```

Reply:
@dwnusbaum it would not fix the underlying problem. Nevertheless I've found a way to workaround the root problem by explicitly managing the classpath (not a perfect solution though) on the call site.